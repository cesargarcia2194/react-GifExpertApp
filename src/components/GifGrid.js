import React from 'react'
import { GifGrdItem } from './GifGrdItem';
import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifGrid = ({category}) => {
    
    const {data:images, loading} = useFetchGifs(category);
    return (
        <>
            <h3 className="animate__animated animate__fadeInDown">{category}</h3>
            {loading && <p>Loading...</p>}
            <div className="card-grid">
                {
                    images.map(img => (
                        <GifGrdItem 
                            key= {img.id}
                            {...img}
                        />
                    ))
                }
            </div>
        </>
    )
}
