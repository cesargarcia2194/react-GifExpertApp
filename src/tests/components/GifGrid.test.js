import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en el componenete <GifGrid />', () => {
    
    
    test('El componenete debe ser renderizado correctamente', () => {
        useFetchGifs.mockReturnValue({
            data:[],
            loading: true
        });
        const wrapper = shallow(<GifGrid category = "Dragon ball"/>)
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe mostrar items cuando se cargan imagenes useFetchGifs', () => {

        const gifs = [{
            id: 'abc',
            url: 'https://imagescda.com',
            title: 'cualquiera'
        }]
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: true
        })
        const wrapper = shallow(<GifGrid category = "Dragon ball"/>);
        expect(wrapper).toMatchSnapshot();
    })
    
    
})
