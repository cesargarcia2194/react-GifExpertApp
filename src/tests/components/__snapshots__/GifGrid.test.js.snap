// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componenete <GifGrid /> Debe mostrar items cuando se cargan imagenes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeInDown"
  >
    Dragon ball
  </h3>
  <p>
    Loading...
  </p>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="abc"
      key="abc"
      title="cualquiera"
      url="https://imagescda.com"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en el componenete <GifGrid /> El componenete debe ser renderizado correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeInDown"
  >
    Dragon ball
  </h3>
  <p>
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
